### PR TITLE
Fix delete objects permissions bug

### LINKED ISSUE #1256: DeleteObjects request causes a permission issue
I have the following snippet of code which runs fine, however if I rerun again I'll get an exception looks like the DeleteObjects handler is returning a permission denied for reference I'm running as admin user and running 
`Version: 1.7.2`

```
Caused by: com.amazonaws.services.s3.model.AmazonS3Exception: Access Denied.; request: POST http://spark-lakefs.s3.lakefs.example.com  {key=[null]} Hadoop 2.7.4, aws-sdk-java/1.11.655 Linux/5.4.0-1029-aws OpenJDK_64-Bit_Server_VM/25.265-b11 java/1.8.0_265 scala/2.12.10 vendor/Azul_Systems,_Inc. com.amazonaws.services.s3.model.DeleteObjectsRequest; Request ID: 77c52098-b323-449a-9017-bf479ff3a436, Extended Request ID: null, Cloud Provider: AWS, Instance ID: i-0e8ca95771fd48a4b (Service: Amazon S3; Status Code: 403; Error Code: AccessDenied; Request ID: 77c52098-b323-449a-9017-bf479ff3a436; S3 Extended Request ID: 23FD7B7E536BD62A), S3 Extended Request ID: 23FD7B7E5
```

```scala
val repo = "spark-lakefs"
val branch = "master"
val dataPath = s"s3a://${dbutils.secrets.get("development","LAKEFS_ACCESS_KEY")}:${dbutils.secrets.get("development","LAKEFS_SECRET_ACCESS_KEY")}@${repo}/${branch}/parquet/"
val testBranch = ""
case class Test(story: String)

val branchStory = Seq(Test("I live on master again")).toDS
branchStory.write.format("delta").mode("overwrite").parquet(dataPath)
```

full stack
```
	at org.apache.spark.sql.execution.datasources.FileFormatWriter$.write(FileFormatWriter.scala:230)
	at org.apache.spark.sql.execution.datasources.InsertIntoHadoopFsRelationCommand.run(InsertIntoHadoopFsRelationCommand.scala:178)
	at org.apache.spark.sql.execution.command.DataWritingCommandExec.sideEffectResult$lzycompute(commands.scala:116)
	at org.apache.spark.sql.execution.command.DataWritingCommandExec.sideEffectResult(commands.scala:114)
	at org.apache.spark.sql.execution.command.DataWritingCommandExec.doExecute(commands.scala:139)
	at org.apache.spark.sql.execution.SparkPlan.$anonfun$execute$1(SparkPlan.scala:200)
	at org.apache.spark.sql.execution.SparkPlan.$anonfun$executeQuery$3(SparkPlan.scala:252)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:165)
	at org.apache.spark.sql.execution.SparkPlan.executeQuery(SparkPlan.scala:248)
	at org.apache.spark.sql.execution.SparkPlan.execute(SparkPlan.scala:192)
	at org.apache.spark.sql.execution.QueryExecution.toRdd$lzycompute(QueryExecution.scala:158)
	at org.apache.spark.sql.execution.QueryExecution.toRdd(QueryExecution.scala:157)
	at org.apache.spark.sql.DataFrameWriter.$anonfun$runCommand$1(DataFrameWriter.scala:999)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$5(SQLExecution.scala:116)
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:249)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$1(SQLExecution.scala:101)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:836)
	at org.apache.spark.sql.execution.SQLExecution$.withCustomExecutionEnv(SQLExecution.scala:77)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:199)
	at org.apache.spark.sql.DataFrameWriter.runCommand(DataFrameWriter.scala:999)
	at org.apache.spark.sql.DataFrameWriter.saveToV1Source(DataFrameWriter.scala:437)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:421)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:294)
	at org.apache.spark.sql.DataFrameWriter.parquet(DataFrameWriter.scala:884)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$$iw$$iw$$iw$$iw$$iw$$iw.<init>(command-316336471633710:9)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$$iw$$iw$$iw$$iw$$iw.<init>(command-316336471633710:69)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$$iw$$iw$$iw$$iw.<init>(command-316336471633710:71)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$$iw$$iw$$iw.<init>(command-316336471633710:73)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$$iw$$iw.<init>(command-316336471633710:75)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$$iw.<init>(command-316336471633710:77)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read.<init>(command-316336471633710:79)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$.<init>(command-316336471633710:83)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$.<clinit>(command-316336471633710)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$eval$.$print$lzycompute(<notebook>:7)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$eval$.$print(<notebook>:6)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$eval.$print(<notebook>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at scala.tools.nsc.interpreter.IMain$ReadEvalPrint.call(IMain.scala:745)
	at scala.tools.nsc.interpreter.IMain$Request.loadAndRun(IMain.scala:1021)
	at scala.tools.nsc.interpreter.IMain.$anonfun$interpret$1(IMain.scala:574)
	at scala.reflect.internal.util.ScalaClassLoader.asContext(ScalaClassLoader.scala:41)
	at scala.reflect.internal.util.ScalaClassLoader.asContext$(ScalaClassLoader.scala:37)
	at scala.reflect.internal.util.AbstractFileClassLoader.asContext(AbstractFileClassLoader.scala:41)
	at scala.tools.nsc.interpreter.IMain.loadAndRunReq$1(IMain.scala:573)
	at scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:600)
	at scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:570)
	at com.databricks.backend.daemon.driver.DriverILoop.execute(DriverILoop.scala:219)
	at com.databricks.backend.daemon.driver.ScalaDriverLocal.$anonfun$repl$1(ScalaDriverLocal.scala:204)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at com.databricks.backend.daemon.driver.DriverLocal$TrapExitInternal$.trapExit(DriverLocal.scala:773)
	at com.databricks.backend.daemon.driver.DriverLocal$TrapExit$.apply(DriverLocal.scala:726)
	at com.databricks.backend.daemon.driver.ScalaDriverLocal.repl(ScalaDriverLocal.scala:204)
	at com.databricks.backend.daemon.driver.DriverLocal.$anonfun$execute$10(DriverLocal.scala:431)
	at com.databricks.logging.UsageLogging.$anonfun$withAttributionContext$1(UsageLogging.scala:239)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
	at com.databricks.logging.UsageLogging.withAttributionContext(UsageLogging.scala:234)
	at com.databricks.logging.UsageLogging.withAttributionContext$(UsageLogging.scala:231)
	at com.databricks.backend.daemon.driver.DriverLocal.withAttributionContext(DriverLocal.scala:48)
	at com.databricks.logging.UsageLogging.withAttributionTags(UsageLogging.scala:276)
	at com.databricks.logging.UsageLogging.withAttributionTags$(UsageLogging.scala:269)
	at com.databricks.backend.daemon.driver.DriverLocal.withAttributionTags(DriverLocal.scala:48)
	at com.databricks.backend.daemon.driver.DriverLocal.execute(DriverLocal.scala:408)
	at com.databricks.backend.daemon.driver.DriverWrapper.$anonfun$tryExecutingCommand$1(DriverWrapper.scala:653)
	at scala.util.Try$.apply(Try.scala:213)
	at com.databricks.backend.daemon.driver.DriverWrapper.tryExecutingCommand(DriverWrapper.scala:645)
	at com.databricks.backend.daemon.driver.DriverWrapper.getCommandOutputAndError(DriverWrapper.scala:486)
	at com.databricks.backend.daemon.driver.DriverWrapper.executeCommand(DriverWrapper.scala:598)
	at com.databricks.backend.daemon.driver.DriverWrapper.runInnerLoop(DriverWrapper.scala:391)
	at com.databricks.backend.daemon.driver.DriverWrapper.runInner(DriverWrapper.scala:337)
	at com.databricks.backend.daemon.driver.DriverWrapper.run(DriverWrapper.scala:219)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.spark.SparkException: Exception thrown in awaitResult: 
	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:431)
	at org.apache.spark.util.ThreadUtils$.parallelMap(ThreadUtils.scala:400)
	at com.databricks.sql.transaction.directory.DirectoryAtomicCommitProtocol.commitJob(DirectoryAtomicCommitProtocol.scala:274)
	at org.apache.spark.sql.execution.datasources.FileFormatWriter$.write(FileFormatWriter.scala:219)
	at org.apache.spark.sql.execution.datasources.InsertIntoHadoopFsRelationCommand.run(InsertIntoHadoopFsRelationCommand.scala:178)
	at org.apache.spark.sql.execution.command.DataWritingCommandExec.sideEffectResult$lzycompute(commands.scala:116)
	at org.apache.spark.sql.execution.command.DataWritingCommandExec.sideEffectResult(commands.scala:114)
	at org.apache.spark.sql.execution.command.DataWritingCommandExec.doExecute(commands.scala:139)
	at org.apache.spark.sql.execution.SparkPlan.$anonfun$execute$1(SparkPlan.scala:200)
	at org.apache.spark.sql.execution.SparkPlan.$anonfun$executeQuery$3(SparkPlan.scala:252)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:165)
	at org.apache.spark.sql.execution.SparkPlan.executeQuery(SparkPlan.scala:248)
	at org.apache.spark.sql.execution.SparkPlan.execute(SparkPlan.scala:192)
	at org.apache.spark.sql.execution.QueryExecution.toRdd$lzycompute(QueryExecution.scala:158)
	at org.apache.spark.sql.execution.QueryExecution.toRdd(QueryExecution.scala:157)
	at org.apache.spark.sql.DataFrameWriter.$anonfun$runCommand$1(DataFrameWriter.scala:999)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$5(SQLExecution.scala:116)
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:249)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$1(SQLExecution.scala:101)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:836)
	at org.apache.spark.sql.execution.SQLExecution$.withCustomExecutionEnv(SQLExecution.scala:77)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:199)
	at org.apache.spark.sql.DataFrameWriter.runCommand(DataFrameWriter.scala:999)
	at org.apache.spark.sql.DataFrameWriter.saveToV1Source(DataFrameWriter.scala:437)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:421)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:294)
	at org.apache.spark.sql.DataFrameWriter.parquet(DataFrameWriter.scala:884)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$$iw$$iw$$iw$$iw$$iw$$iw.<init>(command-316336471633710:9)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$$iw$$iw$$iw$$iw$$iw.<init>(command-316336471633710:69)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$$iw$$iw$$iw$$iw.<init>(command-316336471633710:71)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$$iw$$iw$$iw.<init>(command-316336471633710:73)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$$iw$$iw.<init>(command-316336471633710:75)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$$iw.<init>(command-316336471633710:77)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read.<init>(command-316336471633710:79)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$.<init>(command-316336471633710:83)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$read$.<clinit>(command-316336471633710)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$eval$.$print$lzycompute(<notebook>:7)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$eval$.$print(<notebook>:6)
	at linea438f3c3f5a04942b1e92c96b88acf3271.$eval.$print(<notebook>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at scala.tools.nsc.interpreter.IMain$ReadEvalPrint.call(IMain.scala:745)
	at scala.tools.nsc.interpreter.IMain$Request.loadAndRun(IMain.scala:1021)
	at scala.tools.nsc.interpreter.IMain.$anonfun$interpret$1(IMain.scala:574)
	at scala.reflect.internal.util.ScalaClassLoader.asContext(ScalaClassLoader.scala:41)
	at scala.reflect.internal.util.ScalaClassLoader.asContext$(ScalaClassLoader.scala:37)
	at scala.reflect.internal.util.AbstractFileClassLoader.asContext(AbstractFileClassLoader.scala:41)
	at scala.tools.nsc.interpreter.IMain.loadAndRunReq$1(IMain.scala:573)
	at scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:600)
	at scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:570)
	at com.databricks.backend.daemon.driver.DriverILoop.execute(DriverILoop.scala:219)
	at com.databricks.backend.daemon.driver.ScalaDriverLocal.$anonfun$repl$1(ScalaDriverLocal.scala:204)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at com.databricks.backend.daemon.driver.DriverLocal$TrapExitInternal$.trapExit(DriverLocal.scala:773)
	at com.databricks.backend.daemon.driver.DriverLocal$TrapExit$.apply(DriverLocal.scala:726)
	at com.databricks.backend.daemon.driver.ScalaDriverLocal.repl(ScalaDriverLocal.scala:204)
	at com.databricks.backend.daemon.driver.DriverLocal.$anonfun$execute$10(DriverLocal.scala:431)
	at com.databricks.logging.UsageLogging.$anonfun$withAttributionContext$1(UsageLogging.scala:239)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
	at com.databricks.logging.UsageLogging.withAttributionContext(UsageLogging.scala:234)
	at com.databricks.logging.UsageLogging.withAttributionContext$(UsageLogging.scala:231)
	at com.databricks.backend.daemon.driver.DriverLocal.withAttributionContext(DriverLocal.scala:48)
	at com.databricks.logging.UsageLogging.withAttributionTags(UsageLogging.scala:276)
	at com.databricks.logging.UsageLogging.withAttributionTags$(UsageLogging.scala:269)
	at com.databricks.backend.daemon.driver.DriverLocal.withAttributionTags(DriverLocal.scala:48)
	at com.databricks.backend.daemon.driver.DriverLocal.execute(DriverLocal.scala:408)
	at com.databricks.backend.daemon.driver.DriverWrapper.$anonfun$tryExecutingCommand$1(DriverWrapper.scala:653)
	at scala.util.Try$.apply(Try.scala:213)
	at com.databricks.backend.daemon.driver.DriverWrapper.tryExecutingCommand(DriverWrapper.scala:645)
	at com.databricks.backend.daemon.driver.DriverWrapper.getCommandOutputAndError(DriverWrapper.scala:486)
	at com.databricks.backend.daemon.driver.DriverWrapper.executeCommand(DriverWrapper.scala:598)
	at com.databricks.backend.daemon.driver.DriverWrapper.runInnerLoop(DriverWrapper.scala:391)
	at com.databricks.backend.daemon.driver.DriverWrapper.runInner(DriverWrapper.scala:337)
	at com.databricks.backend.daemon.driver.DriverWrapper.run(DriverWrapper.scala:219)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.amazonaws.services.s3.model.AmazonS3Exception: Access Denied.; request: POST http://spark-lakefs.s3.lakefs.dev.slicelife.com  {key=[null]} Hadoop 2.7.4, aws-sdk-java/1.11.655 Linux/5.4.0-1029-aws OpenJDK_64-Bit_Server_VM/25.265-b11 java/1.8.0_265 scala/2.12.10 vendor/Azul_Systems,_Inc. com.amazonaws.services.s3.model.DeleteObjectsRequest; Request ID: 77c52098-b323-449a-9017-bf479ff3a436, Extended Request ID: null, Cloud Provider: AWS, Instance ID: i-0e8ca95771fd48a4b (Service: Amazon S3; Status Code: 403; Error Code: AccessDenied; Request ID: 77c52098-b323-449a-9017-bf479ff3a436; S3 Extended Request ID: 23FD7B7E536BD62A), S3 Extended Request ID: 23FD7B7E536BD62A
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1712)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1367)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1113)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:770)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:744)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:726)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:686)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:668)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:532)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:512)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4926)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4872)
	at com.amazonaws.services.s3.AmazonS3Client.deleteObjects(AmazonS3Client.java:2257)
	at shaded.databricks.org.apache.hadoop.fs.s3a.S3AFileSystem.lambda$deleteObjects$8(S3AFileSystem.java:1739)
	at shaded.databricks.org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:322)
	at shaded.databricks.org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:285)
	at shaded.databricks.org.apache.hadoop.fs.s3a.S3AFileSystem.deleteObjects(S3AFileSystem.java:1735)
	at shaded.databricks.org.apache.hadoop.fs.s3a.S3AFileSystem.removeKeys(S3AFileSystem.java:2019)
	at shaded.databricks.org.apache.hadoop.fs.s3a.S3AFileSystem.deleteObjects(S3AFileSystem.java:2080)
	at shaded.databricks.org.apache.hadoop.fs.s3a.S3AFileSystem.deleteMultiple(S3AFileSystem.java:2047)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.databricks.sql.util.FileOperations$.$anonfun$deleteMultiple$1(FileOperations.scala:43)
	at scala.util.Try$.apply(Try.scala:213)
	at com.databricks.sql.util.FileOperations$.deleteMultiple(FileOperations.scala:41)
	at com.databricks.sql.transaction.directory.DirectoryAtomicCommitProtocol.$anonfun$commitJob$8(DirectoryAtomicCommitProtocol.scala:317)
	at com.databricks.sql.transaction.directory.DirectoryAtomicCommitProtocol.$anonfun$commitJob$8$adapted(DirectoryAtomicCommitProtocol.scala:274)
	at org.apache.spark.util.ThreadUtils$.$anonfun$parallelMap$2(ThreadUtils.scala:398)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at org.apache.spark.util.threads.SparkThreadLocalCapturingRunnable.$anonfun$run$1(SparkThreadLocalForwardingThreadPoolExecutor.scala:104)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at org.apache.spark.util.threads.SparkThreadLocalCapturingHelper.runWithCaptured(SparkThreadLocalForwardingThreadPoolExecutor.scala:68)
	at org.apache.spark.util.threads.SparkThreadLocalCapturingHelper.runWithCaptured$(SparkThreadLocalForwardingThreadPoolExecutor.scala:54)
	at org.apache.spark.util.threads.SparkThreadLocalCapturingRunnable.runWithCaptured(SparkThreadLocalForwardingThreadPoolExecutor.scala:101)
	at org.apache.spark.util.threads.SparkThreadLocalCapturingRunnable.run(SparkThreadLocalForwardingThreadPoolExecutor.scala:104)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)


```

### PR DESCRIPTION
**Pull request by itaiad200**
closes #1256

### CODE REVIEW

File: nessie/delete_objects_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package nessie
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+
```
Comment (johnnyaug) [inline nessie/delete_objects_test.go:9]: remove line

File: nessie/delete_objects_test.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+const numOfObjects = 10
+
+func TestDeleteObjects(t *testing.T) {
+	ctx, _, repo := setupTest(t)
+
+	files := make([]string, 0, numOfObjects)
```
Comment (johnnyaug) [inline nessie/delete_objects_test.go:18]: why is this array needed?

File: gateway/operations/deleteobjects.go [hunk 3]
```diff
@@ -15,8 +15,13 @@ import (
 
 type DeleteObjects struct{}
 
-func (controller *DeleteObjects) RequiredPermissions(_ *http.Request, _ string) ([]permissions.Permission, error) {
-	return nil, nil
```
Comment (johnnyaug) [inline gateway/operations/deleteobjects.go:19]: So we never supported delete objects? Or did we allow it for all users?

File: permissions/actions.go [hunk 4]
```diff
@@ -20,6 +20,7 @@ const (
 	WriteObjectAction      = "fs:WriteObject"
 	DeleteObjectAction     = "fs:DeleteObject"
 	ListObjectsAction      = "fs:ListObjects"
+	DeleteObjectsAction    = "fs:DeleteObjects"
```
Comment (ozkatz) [inline permissions/actions.go:23]: It should not be a seperate action. Our authorization is modeled after S3, where deleting is an object level operation that requires a DeleteObject permission

**Review by ozkatz — CHANGES_REQUESTED** [review]

File: nessie/delete_objects_test.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	"strconv"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const numOfObjects = 10
```
Comment (nopcoder) [inline nessie/delete_objects_test.go:13]: can you move this one into the testing scope

File: permissions/actions.go (hunk 4 above)
Comment (ozkatz) [reply permissions/actions.go:23]: This is why the authorization currently happens in the Handle func - we parse and authorize every individual object to ensure the user has DeleteObject permissions for it

File: gateway/operations/deleteobjects.go (hunk 3 above)
Comment (itaiad200) [reply gateway/operations/deleteobjects.go:19]: Never supported.

File: nessie/delete_objects_test.go (hunk 1 above)
Comment (itaiad200) [reply nessie/delete_objects_test.go:9]: Done.

File: nessie/delete_objects_test.go (hunk 2 above)
Comment (itaiad200) [reply nessie/delete_objects_test.go:18]: Removed.

File: permissions/actions.go (hunk 4 above)
Comment (itaiad200) [reply permissions/actions.go:23]: Got it.

File: nessie/delete_objects_test.go (hunk 5 above)
Comment (itaiad200) [reply nessie/delete_objects_test.go]: Done.

**Review by johnnyaug — APPROVED** [review]

**Review by ozkatz — APPROVED** [review]
Great!

**Pull request merged**